Lib/re/__init__.py

```diff
@@ -275,60 +275,17 @@ def findall(pattern, string, flags=0):
     has more than one group.
 
     Empty matches are included in the result."""
-    return _compile(pattern, flags).findall(string)
+    return compile(pattern, flags).findall(string)
 
 def finditer(pattern, string, flags=0):
     """Return an iterator over all non-overlapping matches in the
     string.  For each match, the iterator returns a Match object.
 
     Empty matches are included in the result."""
-    return _compile(pattern, flags).finditer(string)
+    return compile(pattern, flags).finditer(string)
 
 def compile(pattern, flags=0):
-    "Compile a regular expression pattern, returning a Pattern object."
-    return _compile(pattern, flags)
-
-def purge():
-    "Clear the regular expression caches"
-    _cache.clear()
-    _cache2.clear()
-    _compile_template.cache_clear()
-
-
-# SPECIAL_CHARS
-# closing ')', '}' and ']'
-# '-' (a range in character set)
-# '&', '~', (extended character set operations)
-# '#' (comment) and WHITESPACE (ignored) in verbose mode
-_special_chars_map = {i: '\\' + chr(i) for i in b'()[]{}?*+-|^$\\.&~# \t\n\r\v\f'}
-
-def escape(pattern):
-    """
-    Escape special characters in a string.
-    """
-    if isinstance(pattern, str):
-        return pattern.translate(_special_chars_map)
-    else:
-        pattern = str(pattern, 'latin1')
-        return pattern.translate(_special_chars_map).encode('latin1')
-
-Pattern = type(_compiler.compile('', 0))
-Match = type(_compiler.compile('', 0).match(''))
-
-# --------------------------------------------------------------------
-# internals
-
-# Use the fact that dict keeps the insertion order.
-# _cache2 uses the simple FIFO policy which has better latency.
-# _cache uses the LRU policy which has better hit rate.
-_cache = {}  # LRU
-_cache2 = {}  # FIFO
-_MAXCACHE = 512
-_MAXCACHE2 = 256
-assert _MAXCACHE2 < _MAXCACHE
-
-def _compile(pattern, flags):
-    # internal: compile pattern
```

Comment:
Generally, it's a bad idea to refactor things unrelated to the issue.

Reply:
I did not refactor anything unrelated. This is just a poor github diff rendering.

Reply:
Ah, I see that now. My bad!

Reply:
Diff indeed looks larger than it is. By setting `_compile = compile` the diff can be made a bit smaller, but I doubt that is worth it